src/driver/result.rs

```diff
@@ -281,7 +281,11 @@ pub unsafe fn malloc_async(
     num_bytes: usize,
 ) -> Result<sys::CUdeviceptr, DriverError> {
     let mut dev_ptr = MaybeUninit::uninit();
+    #[cfg(feature = "alloc_async")]
     sys::cuMemAllocAsync(dev_ptr.as_mut_ptr(), num_bytes, stream).result()?;
+
+    #[cfg(not(feature = "alloc_async"))]
+    sys::cuMemAlloc_v2(dev_ptr.as_mut_ptr(), num_bytes).result()?;
```

Comment:
Hmm we may have to do more to support this - I'm concerned this might interact poorly with multiple non-blocking streams. The current safe api is built around the assumption of stream ordered semantics, so this could lead to subtle timing bugs

Reply:
This is a synced global operation iiurc, meaning it will wait for all streams to finish + block alloc.

I'm far from 100% confident but the code I wrote with this seems to work correctly. And the semantics for cuda library seem to imply that blockign calls will both wait on all streams + sync op, meaning it's safe ops. It's my understanding at least.

Reply:
I think since all allocs/copies for safe api go on default stream this should be fine. The one place where it may be messed up is using CudaStream which currently creates a non blocking stream https://github.com/coreylowman/cudarc/blob/main/src/driver/safe/core.rs#L226.

Probably if we detect this feature we should create Default stream kind https://github.com/coreylowman/cudarc/blob/main/src/driver/result.rs#L196

Reply:
But `cuMemAlloc_v2` doesn't specify the stream.

My understanding is that it should be behind all streams. I don't see any reference in the docs specifying anything about streams though. Maybe I'm wrong, and if not specified it just means send to default stream. 
Not sure how I could test/confirm this though.

Reply:
Yeah allocation should be fine but other calls (like invoking kernels) could still use other streams. That's where the CudaStream & non blocking would come into play

Reply:
Also since this call is *sync*, the other cuda stream calls, would only cause issues if called from other threads, correct ?

Reply:
Oh wait you mean cuMemAlloc_v2 is sync w.r.t the host? I'll have to double check. If so then yes. If it's still async wrt the host, then you could cause issues even when doing threaded

Reply:
I *think* it's sync. It's not explicitly mentionned in the doc I think, but since all async calls are explicitly marked as async, I assume.